internal/datastore/spanner/spanner.go

```diff
@@ -58,6 +59,11 @@ func NewSpannerDatastore(database string, opts ...Option) (datastore.Datastore,
 		return nil, fmt.Errorf(errUnableToInstantiate, err)
 	}
 
+	setting, found := detectEmulatorSetting()
```

Comment:
couldn't this be another `Option` on the spanner datastore (so that we don't have to read env vars here), and then re-use the implementation from the migrate command for the serve command?